### DISCUSSION TITLE
External Storage multiple different paths

### DISCUSSION THREAD

**Discussion by Aceriz** [nextcloud/all-in-one#1089] [Questions] [answered]
Hello I am just finished setting up AIO on unraid

I am wondering for the addition of multiple different paths to external storage points on my server I have added the 
"NEXTCLOUD_MOUNT"  as a variable key  with my /mnt path .  as the value

I have tried adding multiple variable tabs in the unraid page... but  when I do this I am able to get only one of the paths to add correctly (usually the last one) but can not get the other ones to add within the external storage app page of NC settings

Is this something that I am messing up ? or is this something that Nextcloud AIO is limited to ...

**Comment by szaimen** [comment]
Not sure what you want to achieve in detail here. Everything existing in /mnt on the host (even all mounts) will be accessible inside the Nextcloud container.

**Reply by Aceriz** [reply]
> Not sure what you want to achieve in detail here. Everything existing in /mnt on the host (even all mounts) will be accessible inside the Nextcloud container.

Rather than having nextcloudaio having access to my whole /mnt  .   I am wanting to limit the access to specific shares on my unraid setup. 
this is just a means to limit access within the external storage. 

on my prior install of nextcloud (not the aio)  i was able to map specific external mounting points to each specific share that I wanted to add instead of the /mnt

**Reply by szaimen** [reply]
No this is indeed not possible. However access rights are limited to the userid 33 as documented so this is already a restriction you could make use of...

**Reply by Aceriz** [reply]
> No this is indeed not possible. However access rights are limited to the userid 33 as documented so this is already a restriction you could make use of...

okay thanks for such  a quick reply :)

**Reply by Suskerp** [reply]
@Aceriz If this is still an issue for you, I've just discovered that it's possible to use Bind Mounts if you're on Linux (which I presume you are). I have it working on Unraid. Here's what I've done:

1. Create a single central folder (this will be the only folder you pass through to the Docker container). For example: `/mnt/user/nextcloud/external_storage`
2. Run the following commands:

```bash
 cd /mnt/user/nextcloud/external_storage && mkdir folder1
 sudo mount --bind /path/to/first/folder /mnt/user/nextcloud/external_storage/folder1
 mkdir folder2
 sudo mount --bind /path/to/second/folder /mnt/user/nextcloud/external_storage/folder2
```

Note that Bind Mounts do not persist after reboots (AFAIK), so just like me, you'll probably need to redo those binds after each boot. This can be automated, of course.

In Nexcloud you can just add them as local storage with the following paths:

- /mnt/user/nextcloud/external_storage/folder1

- /mnt/user/nextcloud/external_storage/folder2

**Comment by szaimen** [comment, marked as answer]
Hi, there was a PR now that added some documentation on this: https://github.com/nextcloud/all-in-one/pull/8670

See https://github.com/nextcloud/all-in-one#how-to-make-multiple-host-directories-available-inside-the-nextcloud-container
